### PR TITLE
Move scipy.interpolate profiling script into its own module

### PR DESCRIPTION
**Pull request by mdickinson**
There's a profiling script at the end of `scipy.interpolate.interpolate`, but running this script via `python -m scimath.interpolate.interpolate` produces the following slightly worrying warning about "unpredictable behaviour".

```python
(scimath) mdickinson@mirzakhani scimath % python -m scimath.interpolate.interpolate
/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/runpy.py:127: RuntimeWarning: 'scimath.interpolate.interpolate' found in sys.modules after import of package 'scimath.interpolate', but prior to execution of 'scimath.interpolate.interpolate'; this may result in unpredictable behaviour
  warn(RuntimeWarning(msg))
```

After adding some `print` statements for debugging purposes, it's possible to verify that`python -m scipy.interpolate.interpolate` does indeed execute the code in `scipy/interpolate/interpolate.py` twice. (In this particular case that doesn't have any obviously bad side-effects, but the warning is still worrying, and it's easy to fix.)

~This PR moves the profiling script to its own module. To run the profiling script, run `python -m scipy.interpolate.profile`, which should now be warning-free.~

EDIT: Following discussion, this PR simply deletes the profiling code.

### CODE REVIEW

File: scimath/interpolate/profile.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+# Thanks for using Enthought open source!
+
+"""
+Run timings on functions provided by scimath.interpolate.
+"""
+
+from scimath.interpolate import block_average_above, linear
+
+
+def main():
```
[the remark this answers is not in the document]
Comment (rkern) [reply scimath/interpolate/profile.py:18]: A `benchmarks/` directory right at the repo root?

Or just delete it. It was intended to guide development, which is mostly static now.

[the remark this answers is not in the document]
Comment (rahulporuri) [reply scimath/interpolate/profile.py:18]: so we're just deleting it then?

[the remark this answers is not in the document]
Comment (mdickinson) [reply scimath/interpolate/profile.py]: > so we're just deleting it then?

Yep, it's gone.

**Review by rahulporuri — APPROVED** [review]

**Comment by mdickinson** [comment]
Sigh: replaced `scipy` with `scimath` everywhere in the PR description.

**Pull request merged**